crates/proposer-service/src/vkey_hash.rs

```diff
@@ -0,0 +1,106 @@
+use alloy_primitives::B256;
+use serde::{Deserialize, Serialize};
+use sp1_sdk::HashableKey;
+
+pub type HashU32 = [u32; 8];
+
+/// SP1 verifying key hash.
+#[derive(Clone, Copy, Serialize, Deserialize, PartialEq, Eq, PartialOrd, Ord)]
+#[serde(from = "B256", into = "B256")]
+pub struct VKeyHash(HashU32);
```

Comment:
Added this type so we can use `VKeyHash`es consistently. It is now in a submodule here but really it should go into some sort of common crate so other parts of code base have access to it. There is nothing specific to the proposer service in this module.

Reply:
Maybe you could put it immediately to the `aggkit-prover-types`? There is some `pub type Hash` already there.